lib/Chat/Parser/SystemMessage.php

```diff
@@ -763,10 +763,9 @@ protected function getFileFromShare(Room $room, ?Participant $participant, strin
 				}
 
 				$fullPath = $node->getPath();
-				$pathSegments = explode('/', $fullPath, 4);
 				$name = $node->getName();
 				$size = $node->getSize();
-				$path = $pathSegments[3] ?? $name;
+				$path = "Shared/" . ($userFolder->getRelativePath($fullPath) ?? $name);
```

Comment:
Where does the `Shared/` come from?

Reply:
Yeah I don't think it's right, it's hardcoded. Not sure why but getRelatviePath is getting null here and path is getting set to "Shared/" + name. I was trying to figure out what I'm missing Friday but wasn't successful.

Reply:
What is `$fullPath` and `$userFolder->getPath()` in your case?

This is what happens on a quick test for me:
```
$fullPath = "/admin/files/Talk/IMG_0005.jpg"
$userFolder->getRelativePath($fullPath) = "/Talk/IMG_0005.jpg"
```

which is what I would expected here (as `$pathSegments[3]` was also "/Talk/IMG_0005.jpg" in my case).

Reply:
This is the test case that it's failing for me:
`$fullPath = "/user/files/Shared/different"
$userFolder->getRelativePath($fullPath) = "NULL"`

The path is supposed to "Shared/different" not NULL

Reply:
This "cleanup" is killing way too much time. We'll just close the ticket.